### PR TITLE
refactor(api_keys): use `merchant_id` and `key_id` to query the table

### PR DESCRIPTION
**Pull request by Chethan-rao**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Added merchant_id in addition to key_id as a parameter for making db calls to api_keys table

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

When api request was made to the api_keys table, query was made based on key_id. Even if merchant_id passed doesn't exist, if the key_id existed then it returned the result. But now the query will be made based on both merchant_id and key_id.

## How did you test it?

manually

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by SanchithHegde — APPROVED** [review]

File: crates/api_models/src/api_keys.rs [hunk 1]
```diff
@@ -136,8 +136,8 @@ pub struct UpdateApiKeyRequest {
 pub struct RevokeApiKeyResponse {
     /// The identifier for the API Key.
     #[schema(max_length = 64, example = "5hEEqkgJUyuxgSKGArHA4mWSnX")]
+    pub merchant_id: String,
```
Comment (Abhicodes-crypto) [inline crates/api_models/src/api_keys.rs:139]: please correct the schema here. I believe this `/// The identifier for the API Key.
    #[schema(max_length = 64, example = "5hEEqkgJUyuxgSKGArHA4mWSnX")]` doesn't describe `merchant_id`

**Comment by Narayanbhat166** [comment]
Previous index should be dropped and new index of (merchant_id, key_id) should be created.

File: crates/api_models/src/api_keys.rs (hunk 1 above)
Comment (Abhicodes-crypto) [reply crates/api_models/src/api_keys.rs:139]: @Chethan-rao , @SanchithHegde should merchant_id, key_id be added as primary key here ?
https://github.com/juspay/hyperswitch/blob/f624eb52d61561c365cce21e58b08281d096d904/crates/storage_models/src/api_keys.rs#L7

Comment (Chethan-rao) [reply crates/api_models/src/api_keys.rs:139]: No, key_id is enough to uniquely identify records

**Review by SanchithHegde — APPROVED** [review]

**Review by Abhicodes-crypto — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**